src/nft/components/details/AssetPriceDetails.tsx

```diff
@@ -189,6 +202,15 @@ export const AssetPriceDetails = ({ asset, collection }: AssetPriceDetailsProps)
   const itemsInBag = useBag((s) => s.itemsInBag)
   const addAssetToBag = useBag((s) => s.addAssetToBag)
   const removeAssetFromBag = useBag((s) => s.removeAssetFromBag)
+  const { data: fetchedPriceData } = useQuery(['fetchPrice', {}], () => fetchPrice(), {})
+
+  const USDPrice = useMemo(
```

Comment:
this is the same code in 3 places so should probably be a hook right? and would also return the formatted value from the hook